text/0072-signer-vault-recovery.md

```diff
@@ -0,0 +1,86 @@
+In the next screen the user needs to confirm the phrase by typing it in order (copying/pasting will be disabled), this is to ensure they have stored it somewhere other than their clipboard. Failure to correctly provide the **Secret Phrase** would prevent the keypair from being created and stored in the Vault - **the user is required to confirm their Secret Phrase to initialise the Vault**.  
+Having confirmed the **Secret Phrase**, the user is directed to the home screen with a single keypair in their vault.
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+When the vault is created the user needs to enter a password that will be used to encrypt the vault data in the Chrome Storage API.
+The Signer will generate a 12<sup>[1][unresolved-questions]</sup> word phrase based on the [BIP-0039][bip-39_wiki] wordlists. I propose using the [bip39][bip39_js] NPM package to [generate](https://github.com/bitcoinjs/bip39/blob/200c6a93ba0bf946d982e133a00a2c081240af9a/src/index.js#L139 "generateMnemonic function") the **Secret Phrase** and [convert](https://github.com/bitcoinjs/bip39/blob/5faee2c17b2195f30b03cb125df68c20d7dd584b/src/index.js#L58 "mnemonicToSeed function") it to a seed used to [deterministically generate](https://github.com/dchest/tweetnacl-js/blob/f1ec050ceae0861f34280e62498b1d3ed9c350c6/nacl.js#L1113 "fromSeed function") a keypair.
+
+In the event that the **Vault Password** is lost or forgotten the user can reset the Vault (clearing all data therein) and re-initialise it, entering a new **Vault Password** and using their **Secret Phrase** to recover their keypairs.
+
+## Drawbacks
+
+[drawbacks]: #drawbacks
+
+There is a non-trivial amount of work involved in the implementation as well as the introduction of potential security risks to what is otherwise a very secure setup.  
+It is also another secret for users to manage (in addition to the current **Vault Password**).
+
+In the CEP's current state there would need to be a **Secret Phrase** per keypair which could be an inconvenience for users looking to manage multiple keypairs.  
+See the [future possibilities][future-possibilities] section for a potential enhancement.
+
+## Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This implementation for recovery of keypairs is already being used by many entities e.g. [MetaMask][metamask].
+Alternative methods used in non-blockchain services such as **email** or **phone number** may not be as acceptable to a community who value anonymity.
+
+Another alternative would be to actively and strongly encourage users to download their Signer generated keys immediately after creation. There could be an option to 'Skip' but it would be made clear that this is at the user's own risk. Something similar exists in [MetaMask][metamask] but applied to the backup of the Secret Recovery Phrase (since it doesn't support downloading actual key files). 
```

Comment:
MetaMask does allow importing by pasting _private key string_.
![image](https://user-images.githubusercontent.com/5120801/139842093-263f22ec-7b93-4d0a-b6af-a651779ed4cc.png)

Reply:
You can import but I don't believe you can download keys themselves.

Regardless I think we'll be retaining the ability to manually backup key files.